src/engine/filters/index.ts

```diff
@@ -0,0 +1,28 @@
+import {
+  AudienceDefinitionFilter,
+  EngineConditionQuery,
+  PageFeatureResult,
+} from '../../../types';
+import {
+  versionMatches,
+  arrayIntersectsCondition,
+  vectorDistanceCondition,
+  cosineSimilarityCondition,
+} from './conditions';
+
+export const matchesQuery = (
+  query: EngineConditionQuery<AudienceDefinitionFilter>,
+  queryFeatures: PageFeatureResult
+): boolean => {
+  // Can queryFeatures be empty?
```

Comment:
Do you mean can the page views have empty features, yes they can - but in that case they should just be skipped.